File: easyboard-frontend-blank/src/comp/api/posting/command/command/PostingCommand.ts
import { CqrsBaseCommandType, CqrsBaseCommand, NameValueList } from '@nara.drama/prologue';
import { PostingCdo } from '../../api-model';


class PostingCommand extends CqrsBaseCommand {
  postingCdo: PostingCdo | null = null;
  postingCdos: PostingCdo[] = [];
  postingId: string | null = null;
  nameValues: NameValueList | null = null;

  static newRegisterPostingCommand(postingCdo: PostingCdo): PostingCommand {
    const command = new PostingCommand(CqrsBaseCommandType.Register);

    command.postingCdo = postingCdo;
    return command;
  }

  static newRegisterPostingCommands(postingCdos: PostingCdo[]): PostingCommand {
    const command = new PostingCommand(CqrsBaseCommandType.Register);

    command.postingCdos = postingCdos;
    return command;
  }

  static newModifyPostingCommand(postingId: string, nameValues: NameValueList): PostingCommand {
    const command = new PostingCommand(CqrsBaseCommandType.Modify);

    command.postingId = postingId;
    command.nameValues = nameValues;
    return command;
  }

  static newRemovePostingCommand(postingId: string): PostingCommand {
    const command = new PostingCommand(CqrsBaseCommandType.Remove);

    command.postingId = postingId;
    return command;
  }

}

export default PostingCommand;

File: easyboard-backend_blank/easyboard-aggregate/src/main/java/io/naradrama/easyboard/aggregate/posting/api/command/command/PostingCommand.java
/* 
 COPYRIGHT (c) NEXTREE Inc. 2014
 This software is the proprietary of NEXTREE Inc.
 @since 2014. 6. 10.
*/
package io.naradrama.easyboard.aggregate.posting.api.command.command;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import io.naradrama.prologue.domain.cqrs.command.CqrsBaseCommand;
import io.naradrama.easyboard.aggregate.posting.domain.entity.sdo.PostingCdo;
import java.util.List;
import io.naradrama.prologue.domain.NameValueList;
import io.naradrama.prologue.domain.cqrs.command.CqrsBaseCommandType;
import io.naradrama.prologue.util.json.JsonUtil;

@Getter
@Setter
@NoArgsConstructor
public class PostingCommand extends CqrsBaseCommand {
    //
    private PostingCdo postingCdo;
    private List<PostingCdo> postingCdos;
    private boolean multiCdo;
    private String postingId;
    private NameValueList nameValues;

    protected PostingCommand(CqrsBaseCommandType type) {
        //
        super(type);
    }

    public static PostingCommand newRegisterPostingCommand(PostingCdo postingCdo) {
        //
        PostingCommand command = new PostingCommand(CqrsBaseCommandType.Register);
        command.setPostingCdo(postingCdo);
        return command;
    }

    public static PostingCommand newRegisterPostingCommand(List<PostingCdo> postingCdos) {
        //
        PostingCommand command = new PostingCommand(CqrsBaseCommandType.Register);
        command.setPostingCdos(postingCdos);
        command.setMultiCdo(true);
        return command;
    }

    public static PostingCommand newModifyPostingCommand(String postingId, NameValueList nameValues) {
        //
        PostingCommand command = new PostingCommand(CqrsBaseCommandType.Modify);
        command.setPostingId(postingId);
        command.setNameValues(nameValues);
        return command;
    }

    public static PostingCommand newRemovePostingCommand(String postingId) {
        //
        PostingCommand command = new PostingCommand(CqrsBaseCommandType.Remove);
        command.setPostingId(postingId);
        return command;
    }

    public String toString() {
        //
        return toJson();
    }

    public static PostingCommand fromJson(String json) {
        //
        return JsonUtil.fromJson(json, PostingCommand.class);
    }

    public static PostingCommand sampleForRegister() {
        //
        return newRegisterPostingCommand(PostingCdo.sample());
    }

    public static void main(String[] args) {
        //
        System.out.println(sampleForRegister());
    }
}
